packages/blade/src/components/Alert/_decisions/decisions.md

```diff
@@ -0,0 +1,204 @@
+# Alert
+
+Alerts are messages that communicate information to users about any significant changes or explanations inside the system in a **prominent way**.
+
+This document outlines the API of `Alert` component.
+
+<img src="./alerts-thumbnail.png" width="380" alt="" />
+
+## Design
+
+- [Figma - Alerts](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=6824%3A0)
+
+> **Note**
+>
+> The following needs work:
+>
+> 1. color on banking theme in high contrast
+> 2. a11y contrast ratio for notice alerts in high contrast
+> 3. the borderless variant mentioned in full bleed layout shouldn't have any borders and border radii
+
+## API
+
+Annotated component with props:
+
+![Alert breakdown](./alert-breakdown.png)
+
+Sample usage:
+
+```jsx
+import { Alert, Link } from '@razorpay/blade';
+
+<Alert
+  title="International Payments Only"
+  description={
+    <>
+      Currently you can only accept payments in international currencies using PayPal. You cannot
+      accept payments in INR (₹) using PayPal. <Link href="https://razorpay.com">Know More</Link>
+    </>
+  }
```

Comment:
Shouldn't this be a function?
```suggestion
  description={() =>
    <>
      Currently you can only accept payments in international currencies using PayPal. You cannot
      accept payments in INR (₹) using PayPal. <Link href="https://razorpay.com">Know More</Link>
    </>
  }
```

Reply:
I think there isn't a need for that, JSX should be enough. Do you forsee a usecase for passing a function?

Reply:
Not sure. Trying to think if we should have a general guideline for such a thing where a prop would accept JSX or a function. For button and link we do this. I'm worried if consumers would be confused on when to use what

```jsx
<Button icon={CreditCardIcon} />
```

Reply:
I think this should be JSX, because a function isn't needed. If this were a `children` prop we would've expected to pass JSX, expanding same analogy here. Related [composition docs](https://reactjs.org/docs/composition-vs-inheritance.html#containment).

Regarding the point about `icon`. Generally if I pass JSX I would expect it to get rendered without modifications, passing something like `CreditCardIcon` it looks like that might get modified before rendering.

Though if you feel strongly about this approach, let's perhaps discuss and finalize on a call?

Reply:
> Generally if I pass JSX I would expect it to get rendered without modifications, passing something like CreditCardIcon it looks like that might get modified before rendering.

For Icon, it does need to get modified before rendering since the `size` & `color` of the icon depends on the state, intent & variant of button/link. Maybe that could be a decision factor for us - if it needs to be rendered as is then accept JSX otherwise pass it like we pass icon. 

I don't feel strongly about not using JSX here - just concerned that we're taking different approaches in which case we should be aware of what to use when. Lets take it case by case. JSX makes more sense here